libr/bin/p/bin_xnu_kernelcache.c

```diff
@@ -1979,22 +1982,22 @@ static int kernelcache_io_read(RIO *io, RIODesc *fd, ut8 *buf, int count) {
 	r_list_foreach (core->bin->binfiles, iter, bf) {
 		if (bf->fd == fd->fd && bf->bo && bf->bo->bin_obj) {
 			cache = bf->bo->bin_obj;
-			if (pending_bin_files) {
-				RListIter *to_remove = r_list_contains (pending_bin_files, bf);
+			if (cache->pending_bin_files) {
+				RListIter *to_remove = r_list_contains (cache->pending_bin_files, bf);
 				if (to_remove) {
-					r_list_delete (pending_bin_files, to_remove);
-					if (r_list_empty (pending_bin_files)) {
-						r_list_free (pending_bin_files);
-						pending_bin_files = NULL;
+					r_list_delete (cache->pending_bin_files, to_remove);
+					if (r_list_empty (cache->pending_bin_files)) {
+						r_list_free (cache->pending_bin_files);
+						cache->pending_bin_files = NULL;
 					}
 				}
 			}
 			break;
 		}
 	}
 
-	if (!cache) {
-		r_list_foreach (pending_bin_files, iter, bf) {
+	if (!cache && cache->pending_bin_files) {
```

Comment:
i think we have a chicken and egg problem here, if `cache` is null we can't search for it in `cache->pending_bin_files`... that's the reason why this was global